### PR TITLE
feat: scheduled deliveries filter UI

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/4425

### Description:

filter list:
CleanShot 2023-11-02 at 18 11 29@2x

loading state
CleanShot 2023-11-02 at 18 12 04@2x

empty state:
CleanShot 2023-11-02 at 18 12 13@2x

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                        console.info(value);
+                    }}
+                />
+
+                <filterConfig.inputs
+                    filterType={filterType}
+                    field={field}
+                    rule={filter}
+                    onChange={(value) => {
+                        console.info(value);
```
Comment (IrakliJani) [inline packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx:63]: this pr is without state management as you can see

File: packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        <Stack key={filter.id} spacing="xs">
+            <Group spacing="xs">
+                <FieldIcon item={field} />
+                <FieldLabel item={field} hideTableName />
+            </Group>
+
+            <Flex gap="xs">
+                <Select
+                    style={{
+                        flex: '0 0 180px',
```
Comment (IrakliJani) [inline packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx:48]: might change the hardcoded value later

File: packages/frontend/src/features/scheduler/components/SchedulerForm.tsx [hunk 3]
```diff
@@ -164,23 +166,33 @@ const SlackErrorContent: FC<{ slackState: SlackStates }> = ({ slackState }) => {
...
+    resource,
     savedSchedulerData,
     onSubmit,
     onSendNow,
     onBack,
     loading,
     confirmText,
 }) => {
+    const isSchedulerFiltersEnabled =
+        useFeatureFlagEnabled('scheduler-filters');
```
Comment (IrakliJani) [inline packages/frontend/src/features/scheduler/components/SchedulerForm.tsx:194]: am I doing this right? do I need to change anything else?

File: packages/frontend/src/features/scheduler/components/SchedulerModalContent.tsx [hunk 4]
```diff
@@ -109,6 +109,17 @@ const CreateStateContent: FC<{
...
+            resource={
+                isChart
+                    ? {
+                          uuid: resourceUuid,
+                          type: 'chart',
+                      }
+                    : {
+                          uuid: resourceUuid,
+                          type: 'dashboard',
+                      }
```
Comment (IrakliJani) [inline packages/frontend/src/features/scheduler/components/SchedulerModalContent.tsx:121]: I don't like this but I also don't like having `isChart` propagated. does not feel typesafe

File: packages/frontend/src/features/scheduler/components/SchedulerForm.tsx [hunk 5]
```diff
@@ -323,8 +340,12 @@ const SchedulerForm: FC<{
                     <Tabs.Tab value="setup" ml="md">
                         Setup
                     </Tabs.Tab>
+                    {isSchedulerFiltersEnabled && isDashboard ? (
```
Comment (joaoviana) [inline packages/frontend/src/features/scheduler/components/SchedulerForm.tsx:343]: I would even add 
```suggestion
                    {isSchedulerFiltersEnabled && isDashboard && dashboard ? (
```
but not blocking 💪

File: packages/frontend/src/features/scheduler/components/SchedulerForm.tsx (hunk 3 above)
Comment (joaoviana) [reply packages/frontend/src/features/scheduler/components/SchedulerForm.tsx:194]: that's great!

**Review by joaoviana — APPROVED** [review]
let's goooo

File: packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx (hunk 2 above)
Comment (joaoviana) [reply packages/frontend/src/features/scheduler/components/SchedulerFilters.tsx:48]: sounds good

File: packages/frontend/src/features/scheduler/components/SchedulerForm.tsx (hunk 5 above)
Comment (IrakliJani) [reply packages/frontend/src/features/scheduler/components/SchedulerForm.tsx]: good one, made the change

File: packages/frontend/src/features/scheduler/components/SchedulerModalContent.tsx (hunk 4 above)
Comment (joaoviana) [reply packages/frontend/src/features/scheduler/components/SchedulerModalContent.tsx:121]: 🤷 TBH I can't think of a better way than a context and a provider for this 😅

**Pull request merged**